### DISCUSSION TITLE
Off-topic but have you tried Rust?

### DISCUSSION THREAD

**Discussion by TheTechRobo** [DacoTaco/StarStruck#7] [General]
There are  work-in-progress bindings to ogc already made: https://github.com/rust-wii/ogc-rs

(No, I'm not going say you should rewrite everything in Rust, I just thought you might be interested :D)

**Comment by DacoTaco** [comment]
i have sadly never used rust in my life before, but i know it is popular.

this project has nothing to do with libogc btw, if anything it uses devkitARM (the toolkit used for gba/nds/3ds development) and i only use the compiler, since the kernel needs to be as basic as possible.

however, if we were to end up switching to rust, now would be a good time to do so though. most of the code isn't even written yet and most of what is in the repo is just code from mini waiting to be rewritten to be more inline with how ios works.

(hell, the earliest i can release anything of this project to the public is when the kernel is mostly done and i can try and get IOS58 modules to run on it)

there are a few serious questions though : 
* could the project get more attention / people working on it if it was in rust (because this project is huge to do alone and im semi out of my league here haha )
* can devkitARM handle it, since i want to keep this compatible with devkitARM

**Reply by TheTechRobo** [reply]
Again, I wasn't pressuring you to switch to it, just thought you might be interested, since there hasn't been much Wii activity lately (for obvious reasons). 

There are a bunch of people contributing to that repo, and so might be able to help, though. Maybe you could ask them.

Myself, I'm not very good at Wii-dev although during Christmas break I'd like to try to make an Inkball clone in ogc-rs to try to get acquainted, but that won't help much with this project.

**Reply by DacoTaco** [reply]
its a bit off topic, but ive seen some talk about libogc & rust in irc and i think the purpose was there to replace libogc?

either case, what is this no wii activity bs :P
im working on starstruck more than i'd like to admit, which is wii related hehe

i might give rust a go in 2 weeks, i have free time then

**Reply by TheTechRobo** [reply]
I know you've been working on starstruck and there's rc24 and osc etc but I was speaking in a relative term :')
